### PR TITLE
samples: radio_test: Add packet reception limit for start_rx command

### PR DESCRIPTION
**Pull request by maje-emb**
Extended the start_rx command to support receiving a specified number of packets.
After the packets are received or the timeout is exceeded, statistics are displayed.

### CODE REVIEW

**Review by dchat-nordic — CHANGES_REQUESTED** [review]
The code is ok, but I think you should add a note in the https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/samples/peripheral/radio_test/README.html#user_interface to instruct about the optional argument.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -406,6 +406,10 @@ Peripheral samples
   * Added sysbuild configuration for nRF5340.
   * Fixed the dBm value reported for captured frames.
 
+* :ref:`radio_test` sample:
+
+  * Added packet reception limit for start_rx command.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:411]:
```suggestion
  * Added packet reception limit for the ``start_rx`` command.
```

**Review by dchat-nordic — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: samples/peripheral/radio_test/src/radio_test.c [hunk 2]
```diff
@@ -756,6 +767,10 @@ static void radio_rx(uint8_t mode, uint8_t channel, enum transmit_pattern patter
 #endif /* CONFIG_FEM */
 
 	radio_start(true, sweep_processing);
+
+	if (rx_packet_num) {
```
Comment (doki-nordic) [inline samples/peripheral/radio_test/src/radio_test.c:771]: By coding guidelines, we should avoid this. Use `if (rx_packet_num > 0)` or `if (rx_packet_num != 0)`

**Review by doki-nordic — APPROVED** [review]

File: samples/peripheral/radio_test/src/radio_test.c (hunk 2 above)
Comment (maje-emb) [reply samples/peripheral/radio_test/src/radio_test.c]: Good point.

**Pull request merged**